DriveKitDriverDataUI/MySynthesis/Components/SynthesisGauge/MySynthesisGaugeViewModel.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  MySynthesisGaugeViewModel.swift
+//  DriveKitDriverDataUI
+//
+//  Created by Amine Gahbiche on 27/02/2023.
+//  Copyright © 2023 DriveQuant. All rights reserved.
+//
+
+import Foundation
+import DriveKitCoreModule
+import DriveKitCommonUI
+
+class MySynthesisGaugeViewModel {
+    private(set) var scoreType: DKScoreType = .safety
+    private(set) var mean: Double = 0
+    private(set) var min: Double = 0
+    private(set) var max: Double = 0
+    private var score: Double?
+    private(set) var buttonTitle: String = ""
+
+    var hasScore: Bool {
+        return score != nil
+    }
+
+    func configure(scoreType: DKScoreType, mean: Double, min: Double, max: Double, score: Double?) {
+        self.scoreType = scoreType
+        self.mean = mean
+        self.min = min
+        self.max = max
+        self.score = score
+        if let score {
+            self.buttonTitle = DKScoreTypeLevel.getLevel(for: score, scoreType: scoreType).localizedShortDescription()
+        }
+    }
+
+    func getBarItemsToDraw() -> [DKRoundedBarViewItem] {
+        var barViewItems: [DKRoundedBarViewItem] = []
+        let steps = self.scoreType.getSteps()
+
+        guard !steps.isEmpty else {
+            return barViewItems
+        }
+        let max = steps.last!
+        let min = steps.first!
+        for i in 0..<steps.count - 1 {
+            let percent = (steps[i + 1] - steps[i]) / (max - min)
+            let color = ConfigurationCircularProgressView.getScoreColor(value: steps[i + 1], steps: steps)
+            barViewItems.append(DKRoundedBarViewItem(percent: percent, color: color))
+        }
+        return barViewItems
+    }
```

Comment:
```suggestion
    func getBarItemsToDraw() -> [DKRoundedBarViewItem] {
        var barViewItems: [DKRoundedBarViewItem] = []
        let max = DKScoreTypeLevel.excellent.scoreLevels(for: scoreType).upperBound
        let min = DKScoreTypeLevel.veryBad.scoreLevels(for: scoreType).lowerBound
        for scoreLevel in DKScoreTypeLevel.allCases {
            let levelRange = scoreLevel.scoreLevels(for: scoreType)
            let percent = (levelRange.upperBound - levelRange.lowerBound) / (max - min)
            let color = scoreLevel.color
            barViewItems.append(DKRoundedBarViewItem(percent: percent, color: color))
        }
        return barViewItems
    }
```

Tu peux redéfinir cette méthode en réutilisant `scoreLevels(for: scoreType)` comme ceci. Ca évite l'utilisation du `ConfigurationCircularProgressView` et ca réduit un peu le code. Qu'en penses-tu ? (j'ai testé, ca donne exactement le même rendu)

Reply:
corrigé